### PR TITLE
Accept configuration parameters from environment variables

### PR DESCRIPTION
**Pull request by aaronlehmann**
For example, trust_service.type becomes
NOTARY_SERVER_TRUST_SERVICE_TYPE.

Signed-off-by: Aaron Lehmann <aaron.lehmann@docker.com>

### CODE REVIEW

File: cmd/notary-signer/main.go [hunk 1]
```diff
@@ -46,9 +46,11 @@ func init() {
 	// set default log level to Error
 	viper.SetDefault("logging", map[string]interface{}{"level": 2})
 
-	viper.SetEnvPrefix(_EnvPrefix)
-	viper.BindEnv(_DefaultAliasEnv)
-	viper.BindEnv(_PINCode)
+	viper.SetEnvPrefix(envPrefix)
+	viper.SetEnvKeyReplacer(strings.NewReplacer(".", "_"))
+	viper.AutomaticEnv()
+	viper.BindEnv(defaultAliasEnv)
```
Comment (endophage) [inline cmd/notary-signer/main.go]: Don't need to bind if you've done AutomaticEnv()

**Comment by endophage** [comment]
As you added AutomaticEnv to signer it would be nice to remove the BindEnv calls just for cleanliness. Otherwise LGTM.

**Comment by aaronlehmann** [comment]
Removed the BindEnv calls. Thanks. For some reason I thought they were binding config parameters to specific environment variable names. I guess I wasn't looking closely enough.

**Comment by diogomonica** [comment]
@aaronlehmann rebase please

**Comment by aaronlehmann** [comment]
Rebased

**Pull request merged**